reader.js

```diff
@@ -7,15 +9,156 @@ const CLOSE_TIMEOUT = 10000;
 
 let WSS;
 
-function init(wss) {
+function init(server) {
     console.log('Initializing nasonov-reader');
+    const wss = new WebSocket.Server({server});
 
     WSS = wss;
     wss.on('connection', function connected(ws, req) {});
 
+    server.on('request', (request, response) => {connectToInflux(request, response)});
+
     connectToWriter();
 }
 
+function connectToInflux(request, response) {
+
+    if(request.method === 'GET') {
+
+        let requestQuery = new URL(request.url, 'https://habmc.stanfordssi.org/');
+
+        if(requestQuery.pathname.search('/index') > 0) {
+
+            let mission = requestQuery.pathname.substring(1, requestQuery.pathname.search('/index'));
+
+            if (isNaN(mission)) {
+                return;
```

Comment:
We should never return without a response, because that means the connection will stay open forever, wasting resources. You could do something like `response.writeHead(400);` to give an appropriate status code (https://www.restapitutorial.com/httpstatuscodes.html), and then `response.end(JSON.stringify({ error: 'No mission provided' }))`.